platform/android/audio_driver_oboe.h

```diff
@@ -0,0 +1,140 @@
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#ifndef AUDIO_DRIVER_OBOE_H
+#define AUDIO_DRIVER_OBOE_H
+
+#include "core/os/mutex.h"
+#include "servers/audio_server.h"
+
+#include <oboe/Oboe.h>
+
+class AudioDriverOboe : public AudioDriver {
+	enum {
+		BUFFER_SIZE = 1024 * 8, // Burst size shouldn't be more than 1024 and the max channel count supported by Godot is 8.
+		OPENSL_MIX_RATES_SIZE = 10,
+	};
+
+	static const int OPENSL_MIX_RATES[OPENSL_MIX_RATES_SIZE];
+
+	class OutputErrorCallback : public oboe::AudioStreamErrorCallback {
+	public:
+		virtual void onErrorAfterClose(oboe::AudioStream *p_audio_stream, oboe::Result p_error) override;
+	};
+
+	class InputErrorCallback : public oboe::AudioStreamErrorCallback {
+	public:
+		virtual void onErrorAfterClose(oboe::AudioStream *p_audio_stream, oboe::Result p_error) override;
+	};
+
+	class OutputDataCallback16 : public oboe::AudioStreamDataCallback {
+		int32_t buffer[BUFFER_SIZE];
+
+	public:
+		virtual oboe::DataCallbackResult onAudioReady(oboe::AudioStream *p_audio_stream, void *p_audio_data, int32_t p_frames) override;
+	};
+
+	class OutputDataCallback32 : public oboe::AudioStreamDataCallback {
+	public:
+		virtual oboe::DataCallbackResult onAudioReady(oboe::AudioStream *p_audio_stream, void *p_audio_data, int32_t p_frames) override;
+	};
+
+	class InputDataCallback16 : public oboe::AudioStreamDataCallback {
+	public:
+		virtual oboe::DataCallbackResult onAudioReady(oboe::AudioStream *p_audio_stream, void *p_audio_data, int32_t p_frames) override;
+	};
+
+	class InputDataCallback32 : public oboe::AudioStreamDataCallback {
+	public:
+		virtual oboe::DataCallbackResult onAudioReady(oboe::AudioStream *p_audio_stream, void *p_audio_data, int32_t p_frames) override;
+	};
+
+	friend class OutputErrorCallback;
+	friend class InputErrorCallback;
+
+	friend class OutputDataCallback16;
+	friend class OutputDataCallback32;
+
+	friend class InputDataCallback16;
+	friend class InputDataCallback32;
+
+	static AudioDriverOboe *singleton;
+
+	bool active = false;
+	Mutex mutex;
+
+	bool pause = false;
+
+	std::shared_ptr<oboe::AudioStream> audio_output;
+	std::shared_ptr<oboe::AudioStream> audio_input;
+
+	std::shared_ptr<OutputErrorCallback> output_error_callback;
+	std::shared_ptr<InputErrorCallback> input_error_callback;
+
+	std::shared_ptr<oboe::AudioStreamDataCallback> audio_output_callback;
+	std::shared_ptr<oboe::AudioStreamDataCallback> audio_input_callback;
```

Comment:
We don't use the standard library normally, see [here](https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html#standard-template-library), would need to be discussed here depending on the library using it, no other engine component uses this

Reply:
Yeah, I know. Oboe uses shared_ptr and regular pointers, but methods accepting regular pointers are marked as deprecated, when I think about it again, it probably will be fine to use regular pointers, it just might create problems when closing the app (Through is this important?)

Reply:
Will test latter if something is printed in the console, if variables are just pointers

Reply:
Checked when shred_ptr was added, it was in the [google/oboe#821](https://github.com/google/oboe/pull/821) to fix an issue with the data race, without shared_ptr the data race will still occur, so I think shared_ptr should stay